includes/Checker/Default_Check_Repository.php

```diff
@@ -44,7 +44,13 @@ class Default_Check_Repository implements Check_Repository {
 	 */
 	public function register_check( $slug, Check $check ) {
 		if ( ! $check instanceof Runtime_Check && ! $check instanceof Static_Check ) {
-			throw new Exception( __( 'Check must be an instance of Runtime_Check or Static_Check.', 'plugin-check' ) );
+			throw new Exception(
+				sprintf(
+					/* translators: %s: The Check slug. */
+					__( 'Check with slug "%s" must be an instance of Runtime_Check or Static_Check.', 'plugin-check' ),
+					$slug
+				)
+			);
```

Comment:
This change is somewhat unrelated, but it adds the slug to the message to make the exception message more helpful. Without it, you may not know which check caused the exception.